docs/plugins/private-plugins.md

```diff
@@ -122,54 +111,72 @@ Follow the steps below to complete this process:
+:::note
+Only plugins that are properly structured within the `plugins` folder of your connected Git repository will be detected.
+:::
 
-Community Plugin Examples:
+![Publish Your Platform Console Changes](./assets/private-plugins/add-new-plugin.png)
 
-- Browse the [official plugins repository](https://github.com/amplication/plugins/tree/master/plugins) for real-world examples
+## Configure Your Plugin
 
-:::note
-Study the published Node.js and .NET community plugins for best practices and guidance on how to structure your code effectively.
+After adding your plugin, you'll need to configure its settings:
+
+1. **Plugin ID**: Automatically set based on your plugin's folder name in the Git repository. This can only be modified by renaming the folder and re-adding the plugin.
+2. **Display Name**: The human-readable name for your plugin that appears in the UI. By default, it matches the Plugin ID but can be customized for better clarity.
+3. **Icon**: Choose from a variety of technology-specific icons to help visually identify your plugin. You can also customize the icon's color to match your categorization system.
+4. **Code Generator**: Select the appropriate code generator for your plugin:
+  - Node.js
+  - .NET
+  - Blueprints (for custom blueprint solutions)
+5. **Blueprint Compatibility**: Optionally restrict your plugin to work with specific blueprints by selecting them from the dropdown menu.
+6. **Description**: Provide a detailed description of your plugin's purpose. A detailed description helps developers quickly understand and choose the right plugin for their needs.
+
+:::tip
+A meaningful display name, icon, and description will help developers quickly identify and understand your plugin's purpose when browsing the plugin list in services and templates.
 :::
 
-## Add Versions to Your Plugin
+![Configure Your Private Plugin's Settings](./assets/private-plugins/configure-plugin.png)
 
-After developing your plugin, you'll need to version it both in Amplication and your Git repository.
-First, let's add a version through the Amplication UI:
+## Develop Your Plugin
 
-1. Navigate to the Private Plugins tab and select your plugin
-2. Scroll to **Add New Version** and click `Add Version`
-3. Note the version number you assign (e.g., `0.1.0`)
+Now that your plugin structure is set up, it's time to start building your plugin.
 
-![Publish Your Platform Console Changes](./assets/private-plugins/add-new-version.png)
+First, enable dev versions by toggling the `Enable Dev Version` into the on state.
 
-## Tag Your Git Repository
+![Enable Dev Versions](./assets/private-plugins/enable-dev-versions.png)
 
-Next, let's add a git tag to your plugin.
-Amplication uses Git tags to identify plugin versions during code generation.
+Dev Versions streamline the plugin development process by automatically using the latest code from your repository's base branch.
+This eliminates the need to publish new versions during development and testing.
```

Comment:
The "dev" version must be specifically selected in the plugin installation settings to use the dev version.

Reply:
@yuval-hazaz If I'm understanding you correctly, the plugin installation settings page, should have a setting to use the development version?

This page: 
<img width="1172" alt="Screen Shot 2024-12-18 at 7 58 41 PM" src="https://github.com/user-attachments/assets/4ba82c14-b043-40ea-ab4e-45f2d351a158" />

This is separate from the dev versions toggle that developers have to toggle on the plugin repository private plugin settings page?

<img width="1173" alt="Screen Shot 2024-12-18 at 7 59 53 PM" src="https://github.com/user-attachments/assets/d315255f-3975-4eff-9403-c5152d692b4c" />